conjur/logic/hostfactory_logic.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+"""
+VariableLogic module
+
+This module is the business logic for executing the HostFactory commands
+"""
+
+# Internals
+import json
+
+from conjur.data_object.create_token_data import CreateTokenData
+from conjur.errors import MissingRequiredParameterException
+
+
+# pylint: disable=too-few-public-methods
+class HostFactoryLogic:
+    """
+    HostFactoryLogic
+
+    This class holds the business logic for executing and manipulating
+    returned data
+    """
+
+    def __init__(self, client):
+        self.client = client
+
+    def create_token(self, create_token_data: CreateTokenData) -> str:
+        if create_token_data is None:
+            raise MissingRequiredParameterException('create_token_data')
```

Comment:
This is being validate in three different places.. with only to possible paths (sdk going to the api, and cli going controller-> logic -> api) maybe we can skip this validation?

Reply:
This is the case now it can change I prefer not to make this assumption